src/Type/PostObject/Connection/PostObjectConnectionDefinition.php

```diff
@@ -98,29 +99,33 @@ public static function connection( $post_type_object ) {
 		 *
 		 * @since 0.0.5
 		 */
-		return self::$connection[ $post_type_object->name ];
+		return self::$connection[ $from_type ][ $post_type_object->name ];
 
 	}
 
 	/**
 	 * Returns the $args that should be added to the connection args
+	 * @param string $connection
 	 * @return array
 	 */
-	protected static function added_args() {
+	protected static function added_args( $connection ) {
 
 		if ( null === self::$added_args ) {
+			self::$added_args = [];
```

Comment:
Again, does it make sense to just give `self::$added_args` a default value of `[]` so we don't have to do it here?